### PR TITLE
feat(spans): Groundwork for indexed spans tests

### PR DESCRIPTION
**Pull request by wmak**
- This sets up the groundwork for writing more indexed spans tests
- Adds a new `SpansIndexedTestCase` that needs to be inherited so that spans can be stored
- Something to watch out for is that there must be a delay between sending the span and when tests are run, there's a polling function to wait until the span shows up with a 5s timeout
  - In addition instead of making the individual `store_span` function public. The public way to store spans is the bulk `store_spans` function that waits once at the end to hopefully avoid someone calling `store_span` repeatedly (with an up to 5s delay each time)
- Also added a docstring on how to run this locally cause it won't work out of the box until we enable `ENABLE_SPANS_CONSUMER` by default

### CODE REVIEW

**Review by shruthilayaj — APPROVED** [review]

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -2875,3 +2876,73 @@ def _get_path_functions(self):
...
+    base_span = {
+        "is_segment": False,
+        "retention_days": 90,
+        "tags": {},
+        "sentry_tags": {},
+        "measurements": {},
+    }
+
+    # Snuba consumers need a second to process and the consumer needs another second, and another second just to be safe
+    sleep_time = 3
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/testutils/cases.py]: Updated with a polling function instead, runs in about the same time if we only poll for the last event. If this turns out to be insufficent (ie. a random span fails to store) I'll update the polling function to query all the spans instead, went with a single span for performance (querying 1 vs n)

**Comment by wmak** [comment]
Tests failing until https://github.com/getsentry/snuba/pull/5176 merges

**Review by evanh — COMMENTED** [review]
Did you try to use the eventstream endpoint in Snuba for this already? Or are we moving off that endpoint in tests?

Using that would mean the messages would get written directly, and you wouldn't need the polling architecture.

**Comment by wmak** [comment]
> Did you try to use the eventstream endpoint in Snuba for this already? Or are we moving off that endpoint in tests?

@nikhars requested that I add spans this way in tests

**Comment by wmak** [comment]
@evanh :+1: agreed, from DMs with Nikhar I'm going to try reimplementing this but with eventstream

**Comment by wmak** [comment]
Closing this in favour of https://github.com/getsentry/sentry/pull/61766

**Pull request closed without merging**